backend/src/halo2_impl.rs

```diff
@@ -63,8 +66,8 @@ impl<T: FieldElement> BackendImpl<T> for Halo2Mock {
         fixed: &[(&str, Vec<T>)],
         witness: &[(&str, Vec<T>)],
         prev_proof: Option<Proof>,
-        _output_dir: &Path,
-    ) -> io::Result<()> {
+        _output_dir: Option<&Path>,
+    ) -> io::Result<Option<Proof>> {
```

Comment:
How about adding an associated type `Proof` on `BackendImpl` and set it to `()` for mock and return that? It would simplify the trait and give more compile time guarantees